warden/group/manager_memory.go

```diff
@@ -18,6 +19,8 @@ type MemoryManager struct {
 	sync.RWMutex
 }
 
+var _ Manager = (*MemoryManager)(nil)
```

Comment:
is this to assert that `MemoryManager` implements `Manager`?

Reply:
Yes. It will throw a compile time error if the interface isn't satisfied.

Reply:
Since it's used somewhere in the code, that should happen anyways, but it probably makes sense when testing only this specific package/file - maybe move this line to the tests?

Reply:
Your call. I like putting it right after type definitions so I can see how it is intended to be used. Also, I think it was manager_http that didn't fail the build at all if it didn't satisfy Manager. However, the tests failed to build. This didn't feel right to me. Whatever you want though.

Reply:
Makes sense, let's keep it where it is